delta-app/src/main/java/io/cdap/delta/store/DraftService.java

```diff
@@ -388,34 +412,36 @@ private TableAssessmentResponse assessTable(SourceTable sourceTable, TableRegist
 
     StandardizedTableDetail standardizedDetail = tableRegistry.standardize(filteredDetail);
 
-    ColumnRenameInfo columnRenameInfo;
+    ColumnRenameInfo columnRenameInfo = new DefaultRenameInfo(Collections.emptyMap());
+    List<Problem> transformationIssues = new ArrayList<>();
     if (!transformations.isEmpty()) {
       try {
-        DefaultMutableRowSchema rowSchema = TransformationUtil.transformSchema(standardizedDetail.getSchema(),
+        DefaultMutableRowSchema rowSchema = TransformationUtil.transformSchema(table, standardizedDetail.getSchema(),
                                                                                transformations);
        standardizedDetail = new StandardizedTableDetail(standardizedDetail.getDatabase(),
                                                         standardizedDetail.getTable(),
                                                         standardizedDetail.getPrimaryKey(),
                                                         rowSchema.toSchema());
        columnRenameInfo = rowSchema.getRenameInfo();
-      } catch (Exception e) {
-        throw new RuntimeException(String.format("Failed to apply transformations on the schema : %s",
-                                                 e.getMessage()), e);
+      } catch (TransformationException e) {
+        transformationIssues.add(new Problem("Transformation failed",
+                    String.format("Failed to apply transformations on the schema for the table : %s and column : %s " +
+                                    "with error : %s", e.getTableName(), e.getColumnName(), e.getMessage()),
+                    "Please ensure the applied transformation is valid",
+                    "The job cannot be deployed with invalid transformations"));
       }
-    } else {
-      columnRenameInfo = new DefaultRenameInfo(Collections.emptyMap());
     }
     TableAssessment targetAssessment = targetTableAssesor.assess(standardizedDetail);
 
-    return merge(srcAssessment, targetAssessment, columnRenameInfo);
+    return merge(srcAssessment, targetAssessment, columnRenameInfo, transformationIssues);
   }
 
   /**
    * Merge the assessment from the source and target into a single assessment.
    * This amounts to merging the assessment for each column.
    */
   private TableAssessmentResponse merge(TableAssessment srcAssessment, TableAssessment targetAssessment,
-                                        ColumnRenameInfo columnRenameInfo) {
+                                        ColumnRenameInfo columnRenameInfo, List<Problem> transformationProblem) {
```

Comment:
better to be consistent with naming `transformationIssues` vs `transformationProblem`